File: app/src/main/java/com/yashovardhan99/healersdiary/Adapters/PatientPaymentLogsAdapter.java
package com.yashovardhan99.healersdiary.Adapters;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.yashovardhan99.healersdiary.Objects.PaymentSnapshot;
import com.yashovardhan99.healersdiary.R;

import java.util.ArrayList;

/**
 * Created by Yashovardhan99 on 25-06-2018 as a part of HealersDiary.
 */
public class PatientPaymentLogsAdapter extends RecyclerView.Adapter<PatientPaymentLogsAdapter.ViewHolder> {

    private static ArrayList<PaymentSnapshot> paymentSnapshots;

    static class ViewHolder extends RecyclerView.ViewHolder{
        TextView mDateView;
        TextView mAmountView;
        public ViewHolder(RelativeLayout v) {
            super(v);
            mDateView = v.findViewById(R.id.paymentDate);
            mAmountView = v.findViewById(R.id.paymentAmount);
        }
    }

    public PatientPaymentLogsAdapter(ArrayList<PaymentSnapshot> mPaymentSnapshots){ paymentSnapshots = mPaymentSnapshots;}

    @NonNull
    @Override
    public PatientPaymentLogsAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        RelativeLayout v = (RelativeLayout) (LayoutInflater.from(parent.getContext())
                .inflate(R.layout.payment_logs_recycler, parent, false));
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.mAmountView.setText(paymentSnapshots.get(position).getAmount());
        holder.mDateView.setText(paymentSnapshots.get(position).getDate());
    }

    @Override
    public int getItemCount() {
        if(paymentSnapshots!=null)
            return paymentSnapshots.size();
        else
            return 0;
    }

}

File: app/src/main/java/com/yashovardhan99/healersdiary/adapters/PatientPaymentLogsAdapter.kt
package com.yashovardhan99.healersdiary.adapters

import android.content.Context
import android.view.ContextMenu
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.yashovardhan99.healersdiary.R
import com.yashovardhan99.healersdiary.objects.PaymentSnapshot
import java.util.*

/**
 * Created by Yashovardhan99 on 25-06-2018 as a part of HealersDiary.
 */
class PatientPaymentLogsAdapter(private val paymentSnapshots: ArrayList<PaymentSnapshot>) : RecyclerView.Adapter<PatientPaymentLogsAdapter.ViewHolder>() {

    private lateinit var context: Context

    inner class ViewHolder(v: RelativeLayout) : RecyclerView.ViewHolder(v), View.OnCreateContextMenuListener {
        val mDateView: TextView = v.findViewById(R.id.paymentDate)
        val mAmountView: TextView = v.findViewById(R.id.paymentAmount)

        init {
            v.setOnCreateContextMenuListener(this)
        }

        override fun onCreateContextMenu(menu: ContextMenu?, v: View?, menuInfo: ContextMenu.ContextMenuInfo?) {
            menu?.setHeaderTitle(context.getString(R.string.paid_on_date, mAmountView.text, mDateView.text))
            menu?.add(adapterPosition, 0, 0, R.string.edit)
            menu?.add(adapterPosition, 1, 0, R.string.delete)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context)
                .inflate(R.layout.payment_logs_recycler, parent, false) as RelativeLayout
        context = parent.context
        return ViewHolder(v)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.mAmountView.text = paymentSnapshots[position].amount
        holder.mDateView.text = paymentSnapshots[position].date
    }

    override fun getItemCount(): Int {
        return paymentSnapshots.size
    }
}
